src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -492,36 +574,23 @@ private void InheritNullableStateOfFieldOrProperty(int targetContainerSlot, int
                     int targetMemberSlot = GetOrCreateSlot(fieldOrProperty, targetContainerSlot);
                     if (targetMemberSlot >= this.State.Capacity) Normalize(ref this.State);
 
+                    bool? value = !fieldOrPropertyType.IsNullable;
                     if (isByRefTarget)
                     {
-                        // This is a property/field acesses through a by ref entity and it isn't considered declared as not-nullable. 
+                        // This is a property/field access through a by ref entity and it isn't considered declared as not-nullable. 
                         // Since reference can point to the heap, we cannot assume the property/field doesn't have null value after this assignment,
                         // regardless of what value is being assigned.
-                        this.State[targetMemberSlot] = (fieldOrPropertyType.IsNullable == true) ? (bool?)false : null;
                     }
                     else if (valueContainerSlot > 0)
                     {
                         int valueMemberSlot = VariableSlot(fieldOrProperty, valueContainerSlot);
-                        this.State[targetMemberSlot] = valueMemberSlot > 0 && valueMemberSlot < this.State.Capacity ?
-                            this.State[valueMemberSlot] :
-                            null;
+                        value = valueMemberSlot > 0 ? this.State[valueMemberSlot] : null;
```

Comment:
>null [](start = 84, length = 4)

We probably shouldn't set ```value``` to null, but rather leave it unchanged.

Reply:
I can imagine that we haven't allocated a slot for this member in the source container, so we should assume that declaration state is good enough.

---
In reply to: [152698879](https://github.com/dotnet/roslyn/pull/23063#discussion_r152698879) [](ancestors = 152698879)